webhook/webhook.go

```diff
@@ -0,0 +1,119 @@
+package webhook
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"log"
+	"net/http"
+	"time"
+)
+
+var (
+	ErrAlreadyStarted = errors.New("already started")
+)
+
+// HTTPClient represents an interface for the Webhook to send events with.
+type HTTPClient interface {
+	Do(req *http.Request) (*http.Response, error)
+}
+
+// EventLog is the struct that will be send to the Webhook.URL
+type EventLog struct {
+	EventType string    `json:"eventType"`
+	Timestamp time.Time `json:"timestamp"`
+	Event     Event     `json:"event"`
+}
+
+// Webhook can send a Event via POST Request to a specified URL.
+// There are two ways to use a Webhook. You can directly call
+// DispatchEvent or Serve to attach a channel to the Webhook.
+type Webhook struct {
+	stop chan bool
+
+	HTTPClient HTTPClient
+	URL        string
+	Events     []string
```

Comment:
The name is misleading? I first thought you were trying to save the events which had happened as string... But youre using this as filter for the types of events which are allowed to use the webhook.